### PR TITLE
mds: MDSDaemon::init bug fix

### PR DESCRIPTION
**Pull request by renhwztetecs**
MDSDaemon::init bug fix

Signed-off-by: huanwen ren <ren.huanwen@zte.com.cn>

### CODE REVIEW

File: src/mds/MDSDaemon.cc [hunk 1]
```diff
@@ -545,7 +545,6 @@ int MDSDaemon::init(MDSMap::DaemonState wanted_state)
   // Set up admin socket before taking mds_lock, so that ordering
   // is consistent (later we take mds_lock within asok callbacks)
   set_up_admin_socket();
-  g_conf->add_observer(this);
```
Comment (ukernel) [inline src/mds/MDSDaemon.cc]: we should remove the later one. see commit e430f8e4

**Comment by renhwztetecs** [comment]
done!
@ukernel 
thanks for your review

**Comment by jcsp** [comment]
@ukernel are you happy with the updated code?

**Comment by jcsp** [comment]
Tested here http://pulpito.ceph.com/jspray-2016-06-05_19:36:13-fs-wip-jcsp-testing-20160603---basic-mira/

**Pull request merged**